[BUG] Upload to UAT fails

https://github.com/inbo/aspbo/actions/runs/16563789520/job/46843458885 results in:

```
── Error: Check uploaded direct files ──────────────────────────────────────────
<subscriptOutOfBoundsError/error/condition>
Error in `z[["Owner"]][["ID"]]`: subscript out of bounds
Backtrace:
    ▆
 1. └─aws.s3::get_bucket_df(bucket, region = "***")
 2.   ├─base::as.data.frame(r)
 3.   └─aws.s3:::as.data.frame.s3_bucket(r)
 4.     └─base::lapply(...)
 5.       └─aws.s3 (local) FUN(X[[i]], ...)
 6.         └─base::ifelse(is.null(z[["Owner"]]), NA, z[["Owner"]][["ID"]])

Error:
! Test failed
Backtrace:
    ▆
 1. ├─base::source("src/upload_direct_to_UAT.R")
 2. │ ├─base::withVisible(eval(ei, envir))
 3. │ └─base::eval(ei, envir)
 4. │   └─base::eval(ei, envir)
 5. ├─testthat::test_that(...)
 6. │ └─withr (local) `<fn>`()
 7. └─reporter$stop_if_needed()
 8.   └─rlang::abort("Test failed", call = NULL)
Execution halted
Error: Process completed with exit code 1.
```

According to aws.s3 a discussion https://github.com/cloudyr/aws.s3/issues/437 this issue should be fixed by installing aws.s3 v0.3.22. However this action code should install the latest version, in this case v0.3.22. I'm adding a test to see if this is the case.

--- a/src/install_packages_upload_files.R
+++ b/src/install_packages_upload_files.R
@@ -1,8 +1,7 @@
 # get packages installed on machine
 installed <- rownames(installed.packages())
 # specify packages we need
-required <- c(
-  "aws.s3", "magrittr", "readr",
+required <- c("aws.s3", "magrittr", "readr",
   "dplyr", "tidyr", "stringr",
   "testthat"
 )
@@ -15,7 +14,10 @@ if (!all(required %in% installed)) {
 
 # Test if minimum version of aws.s3 is installed
 if (packageVersion("aws.s3") < "0.3.22") {
-  stop("Please update aws.s3 to version 0.3.21 or higher.")
+  install.packages("devtools")
+  cat("aws.s3 version:", paste(unlist(packageVersion("aws.s3")), collapse = "."), "\n
+      ==> installing v0.3.22 from https://rforge.net")
+  devtools::install_version("aws.s3", version = "0.3.22", repos = "https://rforge.net")
 }else{
   cat("aws.s3 version:", paste(unlist(packageVersion("aws.s3")), collapse = "."), "\n")
 }

--- a/src/install_packages_upload_files.R
+++ b/src/install_packages_upload_files.R
@@ -12,3 +12,10 @@ if (!all(required %in% installed)) {
   print(paste("Packages to install:", paste(pkgs_to_install, collapse = ", ")))
   install.packages(pkgs_to_install)
 }
+
+# Test if minimum version of aws.s3 is installed
+if (packageVersion("aws.s3") < "0.3.22") {
+  stop("Please update aws.s3 to version 0.3.21 or higher.")
+}else{
+  cat("aws.s3 version:", paste(unlist(packageVersion("aws.s3")), collapse = "."), "\n")
+}

